learn_physic_model.py

```diff
@@ -0,0 +1,1131 @@
+        x_output = output_node[...,0].t().detach().numpy()
+        y_output = output_node[...,1].t().detach().numpy()
+    plt.figure()
+    plt.plot(x_target, y_target, marker='o', label='Target')
+    plt.plot(x_output, y_output, marker='+', label='Output')
+    plt.legend()
+    plt.title("Position")
+    plt.savefig(path + '/' + name + '.png')
+    plt.close()
+
+"""Set up and generate data
+"""
+
+# Model parameters.
+num_processing_steps_tr = 1
+num_processing_steps_ge = 1
+
+# Data / training parameters.
+batch_size_tr = 256
+batch_size_ge = 100
+num_time_steps = 50
+step_size = 0.1
+num_masses_min_max_tr = (5, 9)
+dist_between_masses_min_max_tr = (0.2, 1.0)
+
+#create the model and the tru physics simulator for data generation
+model = EncodeProcessDecode()
+simulator = SpringMassSimulator(step_size=step_size)
+
+#Base graphs for training
+num_masses_tr = np.random.randint(*num_masses_min_max_tr, size=batch_size_tr)
+dist_between_masses_tr = np.random.uniform(*dist_between_masses_min_max_tr, size= batch_size_tr)
+static_graph_tr = [create_graph(n,d) for n,d in zip(num_masses_tr, dist_between_masses_tr)]
+
+#Base graphs for testing: 4 and 9 nodes
+create_graph_4_ge = [create_graph(4,0.5)]*batch_size_ge
+create_graph_9_ge = [create_graph(9,0.5)]*batch_size_ge
+
+#Generate a training trajectory by adding noise to initial position, spring constants and gravity
+print(">    Generate true trajectories - train ...")
+initial_conditions, true_trajectory_tr_s = [], []
+for elt in static_graph_tr:
+    initial_condition, true_trajectory_tr = generate_trajectory(
+        simulator,
+        elt,
+        num_time_steps,
+        step_size,
+        node_noise_level=0.04,
+        edge_noise_level=5.0,
+        global_noise_level=1.0)
+    initial_conditions.append(initial_condition)
+    true_trajectory_tr_s.append(true_trajectory_tr)
+print(">    ... End.")
+
+print(">    Generate true trajectories - validation ...")
+val_initial_conditions, val_true_trajectory_tr_s = [], []
+for elt in create_graph_4_ge:
+    val_initial_condition, val_true_trajectory_tr = generate_trajectory(
+        simulator,
+        elt,
+        num_time_steps,
+        step_size,
+        node_noise_level=0,
+        edge_noise_level=0,
+        global_noise_level=0)
+    val_initial_conditions.append(initial_condition)
+    val_true_trajectory_tr_s.append(true_trajectory_tr)
+
+for elt in create_graph_9_ge:
+    val_initial_condition, val_true_trajectory_tr = generate_trajectory(
+        simulator,
+        elt,
+        num_time_steps,
+        step_size,
+        node_noise_level=0,
+        edge_noise_level=0,
+        global_noise_level=0)
+    val_initial_conditions.append(initial_condition)
+    val_true_trajectory_tr_s.append(true_trajectory_tr)
+print(">    ... End.")
+
+#Optimizer
+learning_rate = 1e-4
+#init layers with good input
+print(">    Init optimizer ...")
+model(initial_condition, num_processing_steps_tr)
```

Comment:
there is no global variable initial_condition ?

Reply:
No this time it's initial_condition. It's the last value of this variable given l.1020 just to initialize my MLP with good size. I pass it into my model to fixe the input sizes.